pkg/tracing/metadata.go

```diff
@@ -0,0 +1,40 @@
+package tracing
+
+// TODO: PR https://github.com/grpc-ecosystem/grpc-opentracing/ to rework these functions to be reusable
+
+import (
+	"strings"
+
+	"google.golang.org/grpc/metadata"
+)
+
+// metadataReaderWriter satisfies both the opentracing.TextMapReader and
+// opentracing.TextMapWriter interfaces.
+type metadataReaderWriter struct {
+	metadata.MD
+}
+
+func (w metadataReaderWriter) Set(key, val string) {
+	// The GRPC HPACK implementation rejects any uppercase keys here.
+	//
+	// As such, since the HTTP_HEADERS format is case-insensitive anyway, we
+	// blindly lowercase the key (which is guaranteed to work in the
+	// Inject/Extract sense per the OpenTracing spec).
+	key = strings.ToLower(key)
```

Comment:
Sure. This was taken directly from [otgrpc](https://github.com/grpc-ecosystem/grpc-opentracing/blob/master/go/otgrpc/shared.go) and ideally they'll take a PR to rework their code so we don't need to replicate this.